### DISCUSSION TITLE
Tree table

### DISCUSSION THREAD

**Discussion by xiaomlove** [filamentphp/filament#3357] [Help]
I need to display an infinite list of categories without paging, and after retrieving all the data I need to loop through all the data to add a `level` field for each record based on `parent_id`. How do I do this? I see that there is no formatUsing() method in table. Is it possible to customize the data source for a table?

I can see some requirements for this when I search for the `tree` keyword here, but none of the solutions are relevant.

**Comment by danharrin** [comment]
You can probably disable pagination (add https://filamentphp.com/docs/2.x/tables/getting-started#pagination to your List page), and then add a custom table view where you can render the tree https://filamentphp.com/tricks/use-custom-view-for-table-content

**Comment by xiaomlove** [comment]
In fact, I don't need to modify the view, I just need to loop through the entire list to add some fields before rendering

**Reply by fauzie811** [reply]
How did you loop through the list?

**Comment by DevYSM** [comment]
Any updates?
